src/main/java/net/mcreator/preferences/data/HiddenSection.java

```diff
@@ -63,16 +63,17 @@ public class HiddenSection extends PreferencesSection {
 			}
 
 			@Override public JsonElement getSerializedValue() {
-				return PreferencesManager.gson.toJsonTree(value, SortType.class);
+				return PreferencesManager.gson.toJsonTree(value.name());
 			}
 		});
-		java_home = addEntry(new HiddenEntry<>("java_home", null) {
+		java_home = addEntry(new HiddenEntry<File>("java_home", null) {
 			@Override public void setValueFromJsonElement(JsonElement object) {
 				this.value = new File(object.getAsString());
 			}
 
 			@Override public JsonElement getSerializedValue() {
-				return PreferencesManager.gson.toJsonTree(value, File.class);
+				return PreferencesManager.gson.toJsonTree((value == null) ? null : value.getAbsolutePath(),

```

Comment:
No need to convert toJsonTree for simple string. GSON has a concept called JsonPrimitive, use that